src/WebJobs.Script.WebHost/Diagnostics/OptionsFormatter/WebJobsOptionsFactory.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Hosting
+{
+    /// <summary>
+    /// A wrapper factory that allows us to log when creating options.
+    /// </summary>
+    /// <typeparam name="TOptions">The options type to create.</typeparam>
+    internal class WebJobsOptionsFactory<TOptions> : IOptionsFactory<TOptions> where TOptions : class, new()
```

Comment:
The logging wasn't working from the WebHost level (for things like HealthService) as none of these factories/services are registered there. For now, I've just copied them directly from WebJobs as I didn't think we wanted to make all of these classes/interfaces public.

If we'd rather do that, I can make that change as well.